runtime/onert/backend/train/ops/PadLayer.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __ONERT_BACKEND_TRAIN_OPS_PADLAYER_H__
+#define __ONERT_BACKEND_TRAIN_OPS_PADLAYER_H__
+
+#include <backend/IPortableTensor.h>
+#include "OperationUtils.h"
+
+#include <exec/train/ITrainableFunction.h>
+
+namespace onert
+{
+namespace backend
+{
+namespace train
+{
+namespace ops
+{
+
+// Note, this is pad with mode=`CONSTANT`: it doesn't support `REFLECT` and
+// `SYMMETRIC`
+class PadLayer : public ::onert::exec::train::ITrainableFunction
+{
+public:
+  PadLayer();
+
+public:
+  template <typename T> void padImpl(const T *constant_value_data);
+  template <typename T> void depad();
+
+  void configure(const IPortableTensor *input, IPortableTensor *output, const int32_t *padData,
+                 int32_t padRank, const void *constantValueData, IPortableTensor *back_prop_input,
+                 const IPortableTensor *back_prop_output);
+  void forward(bool training) override;
+  void backward() override;
+
+private:
+  const IPortableTensor *_input;
+  IPortableTensor *_output;
+
+  int32_t _padData[8];
```

Comment:
As far as I know, there is no limit to rank of pad layer. How about using `std::vector<int32_t>`?
And it would be good to store pad information by rank. For example, `std::vector<std::pair<int32_t, int32_t>>`.

Reply:
That's a great idea. I'll apply it soon.

Reply:
Hmm now I realize we need a proper compromise. Finally `int32_t *pad` is necessary but `std::pair<>` requires another transformation(eg. `std::vector<std::pair<t,t>>` -> `std::vector<t>`). Therefore, I'll apply just `std::vector<t>` and private function `padRank()`.

Reply:
This would be done separately because cpu::ops::PadLayer code will be fixed.